### ISSUE TITLE
[FEATURE] Multiple local.toml

### ISSUE THREAD

**Issue by SuperCuber** [SuperCuber/dotter#26]
I want to support several local tomls that get read one after another, this is to support a workflow where for example os-level local.tomls are checked into source control but there are still more machine-local variables or secrets that need to be set.

- [x] Consider an option in local.toml to be able to "include" another local.toml
- [x] Document this

**Comment by SuperCuber** [comment]
I feel like this could be also a reason to rethink how `local.toml` works a bit - the mental model (and the algorithm) right now is that each package is patched with the contents of `local.toml` then they are merged.\
This has a bunch of edge cases - what if the package isn't selected? what if it didn't exist in global.toml?\
I think it would be simpler if it worked the other way - first they are merged, then patched.

So a `local.toml` would now look like this:
```toml
packages = ["list", "of", "packages"]
includes = ["first.toml", "second.toml"]

[files]
source = "target"

[variables]
foo = "bar"
```

The algorithm will first take all the packages and merge them in an unspecified order, then it will patch the result with each of the included `local.toml`s in order (they can support includes too I guess?), then it will override it with the final `local.toml`.

- [ ] Make duplicated keys between packages in files or in variables an error when loading configuration

At command-line, I think there's not much point in supporting multiple `local.toml`s - it would just add complexity for not much usefulness.

**Comment by SuperCuber** [comment]
- [ ] Sprinkle `debug!()` and `trace!()`

**Comment by SuperCuber** [comment]
This is a good time to close #21 ... Might just do them together in the same version.

**Comment by SuperCuber** [comment]
To migrate (for #21):
- Move `dotter_settings` to `.dotter`
- Change helper paths accordingly
- Change `included.toml` paths accordingly

**Comment by SuperCuber** [comment]
Closed in 53d5e0ac168eb467e6465d27604240d3cb195bf7, will close once documented
